src/Stratis.Bitcoin.Features.Api/ProofOfStakeConsensusApiFilter.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    /// <summary>
+    /// Swagger Documentation filter for removing mining related methods from API documentation.
+    /// </summary>
+    class ProofOfStakeConsensusApiFilter : IDocumentFilter
+    {
+        public void Apply(SwaggerDocument swaggerDoc, DocumentFilterContext context)
+        {
+            var pathsToRemove = swaggerDoc.Paths
+                .Where(pathItem => pathItem.Key.Contains("/api/Mining/"))
```

Comment:
It would be cool if we can somehow filter by the actual controller name instead of magic strings?

`.Where(pathItem => pathItem.Key.Contains(string.Format("/api/{0}/", typeof(MiningController).Name))`

You will need to somehow filter "controller" out of the name? Maybe `swaggerDoc.Paths` has the controller name it?

Reply:
I tried to do that here, but couldn't get it to work without pulling in a reference to the Mining Feature.